### PR TITLE
Address  stateful-sigs comments in #1650

### PR DESCRIPTION
**Pull request by ducnguyen-sb**
## Add `checked_malloc` and `checked_aligned_alloc`

As mentioned in #1650 , liboqs has a existing pattern `exit(111)` when the malloc allocation failed. This PR add `OQS_MEM_checked_malloc` and `OQS_MEM_checked_aligned_alloc`. 
Two APIs will abruptly exit the program, but *at least* let user know what happens. 
For now, it's close to impossible to propagate errors to callers without a lot of rewrite in the library.

## Use `memcpy` instead of for loop

There is a pattern in `sha2_armv8.c` and `sha2_c.c` that use the for loop to perform an array copy. 
Changed to `memcpy`. This change enables compiler optimizations. 

## Add Stateful signature flags to `CONFIGURE.md`

Fixed minor grammar errors, and add stateful signature flags description.

* [] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
* [] Does this PR change the list of algorithms available -- either adding, removing, or renaming? Does this PR otherwise change an API? (If so, PRs in fully supported downstream projects dependent on these, i.e., [oqs-provider](https://github.com/open-quantum-safe/oqs-provider) and [OQS-OpenSSH](https://github.com/open-quantum-safe/openssh) will also need to be ready for review and merge by the time this is merged.)

### CODE REVIEW

File: src/common/common.h [hunk 1]
```diff
@@ -180,6 +180,59 @@ OQS_API int OQS_MEM_secure_bcmp(const void *a, const void *b, size_t len);
...
+ *       it does not return if the allocation fails, but instead terminates the
+ *       program with an exit status indicating failure.
+ *
+ * @note The memory block returned by this function is not initialized. The caller
+ *       is responsible for initializing the memory if required.
+ *
+ * @note The allocated memory should be freed using the standard `free` function
+ *       when it is no longer needed.
+ */
+OQS_API void *OQS_MEM_checked_malloc(size_t len);
```
Comment (baentsch) [inline src/common/common.h:206]: Is it intentional to designate these new APIs "OQS_API", i.e., making them externally useable? Shouldn't they remain internal?

**Review by baentsch — COMMENTED** [review]
Also thanks for these updates/improvements, @ducnguyen-sb ! Please see single comments.

File: src/common/common.h (hunk 1 above)
Comment (ducnguyen-sb) [reply src/common/common.h:206]: You're right, this should be for internal usage only.

File: src/common/common.h [hunk 2]
```diff
@@ -180,6 +180,59 @@ OQS_API int OQS_MEM_secure_bcmp(const void *a, const void *b, size_t len);
...
+ *       it does not return if the allocation fails, but instead terminates the
+ *       program with an exit status indicating failure.
+ *
+ * @note The memory block returned by this function is not initialized. The caller
+ *       is responsible for initializing the memory if required.
+ *
+ * @note The allocated memory should be freed with `OQS_MEM_aligned_free` when it
+ *       is no longer needed.
+ */
+OQS_API void *OQS_MEM_checked_aligned_alloc(size_t alignment, size_t size);
```
Comment (ducnguyen-sb) [inline src/common/common.h:234]: Changed this function to internal usage only.

Comment (ducnguyen-sb) [reply src/common/common.h]: fixed in https://github.com/open-quantum-safe/liboqs/pull/1656/commits/d28d957b1b1306e26f092f6e08b6c5e83acf6791

File: src/common/common.h (hunk 1 above)
Comment (ducnguyen-sb) [reply src/common/common.h]: fixed in https://github.com/open-quantum-safe/liboqs/pull/1656/commits/d28d957b1b1306e26f092f6e08b6c5e83acf6791

File: CONFIGURE.md [hunk 3]
```diff
@@ -41,21 +41,23 @@ Can be set to the following values:
...
+## OQS_ENABLE_KEM_ALG/OQS_ENABLE_SIG_ALG/OQS_ENABLE_SIG_STFL_ALG
 
-Note: `ALG` in `OQS_ENABLE_KEM_ALG/OQS_ENABLE_SIG_ALG` should be replaced with the specific algorithm name as demonstrated below.
+Note: `ALG` in `OQS_ENABLE_KEM_ALG/OQS_ENABLE_SIG_ALG/OQS_ENABLE_SIG_STFL_ALG` should be replaced with the specific algorithm name as demonstrated below.
 
 This can be set to `ON` or `OFF`, and is `ON` by default. When `OFF`, `ALG` and its code are excluded from the build process. When `ON`, made available are additional options whereby individual variants of `ALG` can be excluded from the build process. 
 
 For example: if `OQS_ENABLE_KEM_BIKE` is set to `ON`, the options `OQS_ENABLE_KEM_bike1_l1_cpa`, `OQS_ENABLE_KEM_bike1_l1_fo`, `OQS_ENABLE_KEM_bike1_l3_cpa`, `OQS_ENABLE_KEM_bike1_l3_fo` are made available (and are set to be `ON` by default). 
 
+To enable `XMSS` stateful signature, set `OQS_ENABLE_SIG_STFL_XMSS` to `ON`, the options `OQS_ENABLE_SIG_STFL_xmss_sha256_h10` and its variants are also set to be `ON` by default. `LMS` stateful signature family can also be enabled by setting `OQS_ENABLE_SIG_STFL_LMS` to `ON`.
```
[the remark this answers is not in the document]
Comment (ducnguyen-sb) [reply CONFIGURE.md]: Ah, it's not intentional, it's my bad, I forgot to update the anchor. 
Updated in https://github.com/open-quantum-safe/liboqs/pull/1656/commits/63cd4ba458435e613fa4c60cd0066b41861e7dd5

**Comment by ducnguyen-sb** [comment]
Is this PR good to merge? @baentsch  @ashman-p  @SWilson4

**Review by ashman-p — APPROVED** [review]

**Review by baentsch — APPROVED** [review]
Looks like an improvement to the current "surprise exit" approach, so good for me. Same caveats as stated in another place, though: Do we want to expose this new API externally? Testing of the new API considered done implicitly by its use in the SHA implementations.

**Comment by ducnguyen-sb** [comment]
@baentsch  Ah, I think these APIs should be used internally.

**Pull request merged**